sdks/python/apache_beam/io/azure/blobstorageio.py

```diff
@@ -0,0 +1,664 @@
+    directory.
+
+    Args:
+      path: Azure Blob Storage file path pattern in the form
+            azfs://<storage-account>/<container>/[name]
+            (ending with a "/").
+
+    Returns:
+      List of tuples of (path, exception), where each path is a blob
+      under the given root. exception is 202 if the operation succeeded
+      or the relevant exception if the operation failed.
+    """
+    assert root.endswith('/')
+
+    # Get the blob under the root directory.
+    paths_to_delete = self.list_prefix(root)
+
+    return self.delete_files(paths_to_delete)
+
+  # We intentionally do not decorate this method with a retry, since the
+  # underlying copy and delete operations are already idempotent operations
+  # protected by retry decorators.
+  def delete_files(self, paths):
+    """Deletes the given Azure Blob Storage blobs from src to dest.
+
+    Args:
+      paths: list of Azure Blob Storage paths in the form
+             azfs://<storage-account>/<container>/[name] that give the
+             file blobs to be deleted.
+
+    Returns:
+      List of tuples of (src, dest, exception) in the same order as the
+      src_dest_pairs argument, where exception is 202 if the operation
+      succeeded or the relevant exception if the operation failed.
+    """
+    if not paths:
+      return []
+
+    # Group blobs into containers.
+    containers, blobs = zip(*[parse_azfs_path(path, get_account=False) \
+        for path in paths])
+
+    grouped_blobs = {container: [] for container in containers}
+
+    # Fill dictionary.
+    for container, blob in zip(containers, blobs):
+      grouped_blobs[container].append(blob)
+
+    results = {}
+
+    # Delete minibatches of blobs for each container.
+    for container, blobs in grouped_blobs.items():
+      for i in range(0, len(blobs), MAX_BATCH_OPERATION_SIZE):
+        blobs_to_delete = blobs[i:i + MAX_BATCH_OPERATION_SIZE]
+        results.update(self._delete_batch(container, blobs_to_delete))
+
+    final_results = \
+        [(path, results[parse_azfs_path(path, get_account=False)]) \
+        for path in paths]
+
+    return final_results
+
+  @retry.with_exponential_backoff(
+      retry_filter=retry.retry_on_beam_io_error_filter)
+  def _delete_batch(self, container, blobs):
+    """A helper method. Azure Blob Storage Python Client allows batch
+    deletions for blobs within the same container.
+
+    Args:
+      container: container name.
+      blobs: list of blobs to be deleted.
+
+    Returns:
+      Dictionary of the form {(container, blob): error}, where error is
+      202 if the operation succeeded.
+    """
+    container_client = self.client.get_container_client(container)
+    results = {}
+
+    try:
+      response = container_client.delete_blobs(
+          *blobs, raise_on_any_failure=False)
+
+      for blob, error in zip(blobs, response):
+        results[(container, blob)] = error.status_code
+
+    except BlobStorageError as e:
```

Comment:
I think you should handle both `BlobStorageError` and `PartialBatchErrorException` on all blob storage operations (PartialBatchErrorException is raised in, for example, https://github.com/Azure/azure-sdk-for-python/issues/13183) -- otherwise, what ends up happening is that only the status code from PartialBatchErrorException is retrieved, but the message is silenced and not logged at all.